FlowCal/plot.py

```diff
@@ -1061,6 +1302,1271 @@ def hist1d(data_list,
+            data_length = len(data)
+        except TypeError:
+            data = [data]
+            data_length = 1
+    else:
+        # assume ND sequence or sequence of ND sequences
+        try:
+            first_element               = next(iter(data))
+            first_element_first_element = next(iter(first_element))
+        except TypeError:
+            msg  = "`data` should be ND sequence or sequence of ND sequences."
+            msg += " Set `channel` to None to use 1D sequence or sequence of"
+            msg += " 1D sequences."
+            raise TypeError(msg)
+
+        # promote singleton if necessary
+        try:
+            iter(first_element_first_element)  # success => sequence of ND sequences
+            data_length = len(data)
+        except TypeError:
+            data = [data]
+            data_length = 1
+
+        # exctract channel
+        try:
+            data = [d[:,channel] for d in data]
+        except TypeError:
+            data = [[row[channel] for row in d] for d in data]
+        if min_data is not None:
+            try:
+                min_data = min_data[:,channel]
+            except TypeError:
+                min_data = [row[channel] for row in min_data]
+        if max_data is not None:
+            try:
+                max_data = max_data[:,channel]
+            except TypeError:
+                max_data = [row[channel] for row in max_data]
+        if logx_zero_data is not None:
+            try:
+                logx_zero_data = logx_zero_data[:,channel]
+            except TypeError:
+                logx_zero_data = [row[channel] for row in logx_zero_data]
+
+    # understand `positions`
+    if positions is None:
+        positions = np.arange(1,data_length+1, dtype=np.float)
+        if xscale == 'log':
+            positions = 10**positions
+        positions_length = len(positions)
+    else:
+        try:
+            positions_length = len(positions)
+        except TypeError:
+            positions = [positions]
+            positions_length = 1
+
+    if positions_length != data_length:
+        msg  = "`positions` must have the same length as `data`"
+        raise ValueError(msg)
+
+    # separately illustrate position=0 if log x-axis
+    if xscale == 'log' and 0 in list(positions):
+        data      = list(data)
+        positions = list(positions)
+
+        zero_idx = [idx
+                    for idx,pos in enumerate(positions)
+                    if pos == 0]
+
+        if len(zero_idx) > 1:
+            msg  = "attempting to separately illustrate position=0 violin,"
+            msg += " but found multiple instances"
+            raise ValueError(msg)
+        zero_idx = zero_idx[0]
+
+        zero_data = data.pop(zero_idx)
+        del positions[zero_idx]
+        data_length      = len(data)
+        positions_length = len(positions)
+
+        if logx_zero_data is None:
+            logx_zero_data = zero_data
```

Comment:
Maybe there needs to be a warning here if `log_zero_data` is not None? In this case, `data[zero_idx]` will be ignored, which may not be immediately obvious to the user.

Reply:
Yeah, I think that's a good idea if we keep this behavior. The silent preference for `log_zero_data` over `data[zero_idx]` could otherwise be dubious.

You originally weren't sure about this automatic separation of the zero violin, though. I offered the alternative of ignoring `data[zero_idx]` if the x-axis is logarithmic, but we never resolved the issue.